### ISSUE TITLE
Textual Entailment demo and performance

### ISSUE THREAD

**Issue by seriouscoder123** [allenai/allennlp#1137]
I've tried to download the model (decomposable-attention-elmo-2018.02.19.tar.gz) and run Textual Entailment prediction locally. There are two issues that I found:

1) The local version returns results that are different from the demo hosted @ demo.allennlp.org. In some cases, the difference is actually quite sizeable.
For example, with this JSONL file:

```
cat examples.jsonl                                                                                                                        
{"hypothesis": "He can influence and develop strong professional relationships", "premise": "John develops effective working relationships"}
```
and command
```
python -m allennlp.run predict decomposable-attention-elmo-2018.02.19.tar.gz examples.jsonl
```
produces the following result:

```
input:  {'hypothesis': 'He can influence and develop strong professional relationships', 'premise': 'John develops effective working relationships'}
prediction:  {"label_logits": [1.1014702320098877, -2.091261386871338, 2.0011751651763916], "label_probs": [0.2857194244861603, 0.011731510050594807, 0.7025490403175354]}
```

If you tried the demo, the results will be:
```
Entailment | 83.1%
Contradiction | 0.1%
Neutral | 16.8%
```
So Entailment returned by command line is 28.6% compared with 83.1% from the demo.

What version of code and model are being used @ demo.allennlp.org? Why does it produces such different results from command line version?

2) I measured performance of Textual Entailment model by loading the archive using ``load_archive`` and run ``Predictor.predict_json``. The execution time of ``predict_json`` method is around 200ms. The server has 28 cores and 110GB of RAM. Is this level of performance normal? I was expecting inference phase to be a lot faster.

Thanks.

**Comment by schmmd** [comment]
@seriouscoder123 that performance seems reasonable.  We have not focused on optimizing AllenNLP.  It won't take advantage of multiple cores or your large amount of RAM, but you should be able to make predictions faster by batching them and running them on the GPU.

That's odd there's a difference.  I'll investigate.

**Comment by matt-peters** [comment]
Running anything with ELMo will be slow on CPU.

The difference may be due to the ELMo biLM states as they need a batch or two to warm up.  To diagnose this, create a  `examples.jsonl` that includes the same sentences say 10 times and see if the predictions are consistent.  Make sure to use `--batch-size 1`` when predicting for this experiment.

**Comment by seriouscoder123** [comment]
Thanks for the replies guys.

@matt-peters I've tried your recommendation but the results seem to stay the same:

```
input:  {'hypothesis': 'He can influence and develop strong professional relationships', 'premise': 'John develops effective working relationships'}
prediction:  {"label_logits": [1.1014702320098877, -2.091261386871338, 2.0011751651763916], "label_probs": [0.2857194244861603, 0.011731510050594807, 0.7025490403175354]}

input:  {'hypothesis': 'He can influence and develop strong professional relationships', 'premise': 'John develops effective working relationships'}
prediction:  {"label_logits": [1.078644037246704, -2.106003761291504, 2.0055251121520996], "label_probs": [0.2802683413028717, 0.011601097881793976, 0.7081305980682373]}

input:  {'hypothesis': 'He can influence and develop strong professional relationships', 'premise': 'John develops effective working relationships'}
prediction:  {"label_logits": [1.0790457725524902, -2.106219530105591, 2.0054049491882324], "label_probs": [0.2803739011287689, 0.011598301120102406, 0.7080277800559998]}

input:  {'hypothesis': 'He can influence and develop strong professional relationships', 'premise': 'John develops effective working relationships'}
prediction:  {"label_logits": [1.079058051109314, -2.1062307357788086, 2.005415678024292], "label_probs": [0.28037431836128235, 0.011598045006394386, 0.7080276608467102]}

input:  {'hypothesis': 'He can influence and develop strong professional relationships', 'premise': 'John develops effective working relationships'}
prediction:  {"label_logits": [1.0790590047836304, -2.106231927871704, 2.0054163932800293], "label_probs": [0.28037434816360474, 0.011598021723330021, 0.7080276012420654]}

input:  {'hypothesis': 'He can influence and develop strong professional relationships', 'premise': 'John develops effective working relationships'}
prediction:  {"label_logits": [1.079059362411499, -2.106231927871704, 2.005415916442871], "label_probs": [0.2803744971752167, 0.011598029173910618, 0.7080274224281311]}

input:  {'hypothesis': 'He can influence and develop strong professional relationships', 'premise': 'John develops effective working relationships'}
prediction:  {"label_logits": [1.0790594816207886, -2.106231927871704, 2.0054163932800293], "label_probs": [0.2803744673728943, 0.011598020792007446, 0.7080275416374207]}

input:  {'hypothesis': 'He can influence and develop strong professional relationships', 'premise': 'John develops effective working relationships'}
prediction:  {"label_logits": [1.07905912399292, -2.106231689453125, 2.005415916442871], "label_probs": [0.2803744673728943, 0.011598030105233192, 0.7080274820327759]}

input:  {'hypothesis': 'He can influence and develop strong professional relationships', 'premise': 'John develops effective working relationships'}
prediction:  {"label_logits": [1.0790598392486572, -2.106232166290283, 2.005415916442871], "label_probs": [0.2803746461868286, 0.011598022654652596, 0.7080273628234863]}

input:  {'hypothesis': 'He can influence and develop strong professional relationships', 'premise': 'John develops effective working relationships'}
prediction:  {"label_logits": [1.0790594816207886, -2.106232166290283, 2.00541615486145], "label_probs": [0.28037452697753906, 0.011598019860684872, 0.7080274820327759]}

input:  {'hypothesis': 'He can influence and develop strong professional relationships', 'premise': 'John develops effective working relationships'}
prediction:  {"label_logits": [1.0790598392486572, -2.1062324047088623, 2.0054163932800293], "label_probs": [0.28037452697753906, 0.011598019860684872, 0.7080274820327759]}

input:  {'hypothesis': 'He can influence and develop strong professional relationships', 'premise': 'John develops effective working relationships'}
prediction:  {"label_logits": [1.079059362411499, -2.106231927871704, 2.00541615486145], "label_probs": [0.2803744673728943, 0.011598024517297745, 0.7080274820327759]}
```

**Comment by seriouscoder123** [comment]
Actually it was my fault. I mixed up "hypothesis" and "premise" in my JSONL file :(.
